### DISCUSSION TITLE
Proposed terminology

### DISCUSSION THREAD

**Discussion by twistedddx** [thirtythreeforty/neolink#105] [General]
Here's a discussion to break the ice on discussions. 

This is my proposed naming of different parts of this technology:

BC TCP - protocol used by powered cameras such as RLC-410, B800, RLN8-410. Typical magic 0abcdef0 or 0fedcba0
BC UDP - protocol used by battery cameras such as Argus. Typical magic 2a87cf3a

BC Modern - messaging for newer (post2016) camera firmware. Many messages include XML. 
BC Legacy - messaging for older (pre 2016) camera firmware. It is mostly C structs and does not use XML. In modern times this messaging is only used to start connection before switching to BC Modern. 

BC Media - file format for audio/video data. https://www.wasteofcash.com/BCConvert/BC_fileformat.txt

BC Discovery - broadcast to discover devices Mac, ip, Port, name and uid. Port 2000/3000
BC Cloud Protocol - Reolink remote connection via AWS. User may use BC UID to connect to device remotely

**Comment by Lanwalker** [comment]
Now explain / define the ' BC ' ahead of all the terms.

**Comment by twistedddx** [comment]
Bc is shorthand for Baichuan aka Shenzhen Baichuan Secruity Technology.
It is the company that created this CCTV product.
It is assumed the company Baichuan now uses the name Reolink when retailing direct to customers. 

The code officially released by Swann and Reolink etc over the years internally refers to itself as BC. 
Eg BC is the manufacturers name that does not directly sell to customers.

**Comment by Lanwalker** [comment]
Thank you, Sir !
I was wondering about who were all the players in the ' game ' and that sounds right... and explains all the catching up in the software and firmware we are seeing too.

Darn it... and I try to BUY UNITED STATES MADE anymore<G>
I guess we have to get the crap off the shelves for other things to be there.
BTW, I admin for the United States Minutemen anymore, you can find us on MeWe.com since FB, Twitter and Youtube has booted us<G>

**Reply by thirtythreeforty** [reply]
@Lanwalker would you mind configuring your email client to omit the signature and quoted reply? It shows up as unformatted text at the bottom of your posts. I've tried to edit it out where I see it, but still.

**Comment by thirtythreeforty** [comment]
All sounds good, except let's not call the remote server variant "P2P" since to everyone except Reolink that means "lack of central server." Maybe "Reolink Cloud protocol"?

**Reply by thirtythreeforty** [reply]
Er, if we're being consistent, "BC cloud protocol"

**Comment by Lanwalker** [comment]
I'll just remove it thru these conversations. Thank you for letting me know it was bothering people. I admin for many groups so that was my normal response so everybody could get to me as needed.
